modules/plugin/geotiff/src/test/java/org/geotools/gce/geotiff/GeoTiffReaderTest.java

```diff
@@ -1471,4 +1471,27 @@ public void testBandDuplication() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testFloatNegInfinityNoData() throws IOException {
```

Comment:
This test fails with the error in the description in the main branch with no extra code changes.